sh_scrapy/env.py

```diff
@@ -13,6 +13,12 @@ def _make_scrapy_args(arg, args_dict):
     return args
 
 
+def get_scrapy_list_args_and_env(msg):
+    args = ['scrapy', 'list']
+    env = {'SCRAPY_PROJECT_ID': msg['project']}
```

Comment:
As there's no job key when getting spiders list, I've added a new field `project` to the message to pass project id. `SCRAPY_PROJECT_ID` itself is important because it's used to run Scrapy or a custom script in the `crawl` module.